src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,6 +32,11 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        String[] addressElements = trimmedAddress.split(",");
+        this.block = new Block(addressElements[0].trim());
+        this.street = new Street(addressElements[1].trim());
+        this.unit = new Unit(addressElements[2].trim());
+        this.postalCode = new PostalCode(addressElements[3].trim());
         this.value = trimmedAddress;
```

Comment:
Is this variable still necessary? Can two addresses have the same block, street, unit, and postalCode, but different trimmedAddress?